service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccess.java

```diff
@@ -110,70 +110,61 @@ public FolderAccess(I_DomainAccess domainAccess, UUID ehrId, I_ContributionAcces
-
-    @Override
-    public Boolean update(Timestamp transactionTime) {
-        // not valid, because missing mandatory data, like committer
-        throw new InternalServerException(CALLED_INVALID_ACCESS_LAYER_METHOD);
-    }
-
-    @Override
-    public Boolean update(final Timestamp transactionTime, final boolean force) {
-        // not valid, because missing mandatory data, like committer
-        throw new InternalServerException(CALLED_INVALID_ACCESS_LAYER_METHOD);
-    }
+    // *************Data Access and modification methods*****************
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public Boolean update(final Timestamp transactionTime, final boolean force, UUID contribution,
-        UUID systemId, UUID committerId, String description, ContributionChangeType changeType) {
+    public boolean update(final LocalDateTime transactionTime, UUID systemId, UUID committerId, String description, ContributionChangeType changeType) {
         /*create new contribution*/
         UUID oldContribution = this.folderRecord.getInContribution();
         UUID newContribution;
 
-        // if no custom contribution is provided create a new one, otherwise use given one
-        if (contribution == null) {
-            UUID contributionAccessEhrId = this.contributionAccess.getEhrId();
-            /*save the EHR id from oldContribution since it will be the same as this is an update operation*/
-            if (this.contributionAccess.getEhrId() == null) {
-                ContributionRecord rec = getContext().fetchOne(CONTRIBUTION, CONTRIBUTION.ID.eq(oldContribution));
-                contributionAccessEhrId = rec.getEhrId();
-            }
-            this.contributionAccess.setEhrId(contributionAccessEhrId);
-
-            this.contributionAccess.commit(transactionTime, committerId, systemId, ContributionDataType.folder, ContributionDef.ContributionState.COMPLETE, I_ConceptAccess.ContributionChangeType.MODIFICATION, description);
-            this.getFolderRecord().setInContribution(this.contributionAccess.getId());
-        } else {
-            this.getFolderRecord().setInContribution(contribution);
+        // No custom contribution is provided, so create a new one
+        UUID contributionAccessEhrId = this.contributionAccess.getEhrId();
+        /*save the EHR id from oldContribution since it will be the same as this is an update operation*/
+        if (this.contributionAccess.getEhrId() == null) {
+            ContributionRecord rec = getContext().fetchOne(CONTRIBUTION, CONTRIBUTION.ID.eq(oldContribution));
+            contributionAccessEhrId = rec.getEhrId();
         }
+        this.contributionAccess.setEhrId(contributionAccessEhrId);
+
+        this.contributionAccess.commit(Timestamp.valueOf(transactionTime), committerId, systemId,
+            ContributionDataType.folder, ContributionDef.ContributionState.COMPLETE, I_ConceptAccess.ContributionChangeType.MODIFICATION, description);
```

Comment:
@jakesmolka is the change type always MODIFICATION here? can't be AMENDMENT?

Reply:
Good one, thanks! I'll commit a change.